### PR TITLE
windows: unify mailbox_proto.h file & added missing queries

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
Add ioctls to get the mailbox & board info.
Move common structures to mailbox_proto.h like in linux
Fix windows device query requests.

Signed-off-by: Rajkumar Rampelli <rajkumar@xilinx.com>

### CODE REVIEW

**Comment by rajkumar-xilinx** [comment]
C:\Users\Administrator\Desktop\raj\XRT\build\WRelease\xilinx\xrt\bin>**xbutil examine -r all**
```
System Configuration
  OS Name              : Windows NT
  Release              : 20215.rs_prerelease.200911-1500
  Version              : 6.3
  Machine              : x86_64
  CPU Cores            : 6
  Memory               : 16382 MB
  Distribution         : Windows Server 2019 Datacenter
  Model                : Virtual Machine

XRT
  Version              : 2.12.0
  Branch               : 23jul_tmp
  Hash                 : 659a957a7fdeaf755120db6ffd8c2f948e64045b
  Hash Date            : 2021-07-23 20:56:39

Devices present
  [0000:00:00.0] : xilinx_u250_xdma_201830_2


-----------------------------------------------
1/1 [0000:00:00.0] : xilinx_u250_xdma_201830_2
-----------------------------------------------
AIE
  <AIE information unavailable>

Memory Information

Xclbin UUID
  2A44FC56-4EBA-430C-95A4-13FA87DEE9CC

Compute Units
  PL Compute Units
    Index   Name                          Base_Address    Usage   Status

  PS Compute Units
    Index   Name                          Base_Address    Usage   Status

Debug IP Status
  INFO: Failed to find any Debug IP Layout section in the bitstream loaded on device. Ensure that a valid bitstream with debug IPs (AIM, LAPC) is successfully downloaded.

 INFO: Debug IP Data is not populated.
 INFO: Debug IP Data is not populated.
Pcie Info
  Vendor                 : 0x10ee
  Device                 : 0x5005
  Sub Device             : 0x000e
  Sub Vendor             : 0x10ee
  PCIe                   : Gen3x16
  DMA Thread Count       : 0
  CPU Affinity           : 0
  Shared Host Memory     : 0 Bytes
  Max Shared Host Memory : 0 Bytes

Platform
  XSA Name               : xilinx_u250_xdma_201830_2
  FPGA Name              : xcu250-figd2104-2L-e
  JTAG ID Code           : 0x4b57093
  DDR Size               : 68719476736 Bytes
  DDR Count              : 4
  Mig Calibrated         : true
  P2P Status             : not supported

Mac Addresses            : 00:0A:35:05:FE:28
                         : 00:0A:35:05:FE:29

Electrical
  Max Power              : 75 Watts
  Power                  : 35.905932 Watts
  Power Warning          : false

  Power Rails            : Voltage   Current
  12 Volts Auxillary     :  0.459 V,  0.016 A
  12 Volts PCI Express   : 12.132 V,  2.959 A
  3.3 Volts PCI Express  :  3.350 V
  3.3 Volts Auxillary    :  3.305 V
  Internal FPGA Vcc      :  0.851 V,  9.500 A
  DDR Vpp Bottom         :  2.500 V
  DDR Vpp Top            :  2.500 V
  5.5 Volts System       :  5.491 V
  Vcc 1.2 Volts Top      :  1.204 V
  Vcc 1.2 Volts Bottom   :  1.206 V
  1.8 Volts Top          :  1.832 V
  0.9 Volts Vcc          :  0.910 V
  12 Volts SW            : 12.129 V
  Mgt Vtt                :  1.204 V

Mailbox
  Total bytes received   : 16448 Bytes
  Unknown                : 0
  Test msg ready         : 0
  Test msg fetch         : 0
  Lock bitstream         : 0
  Unlock bitstream       : 0
  Hot reset              : 0
  Firewall trip          : 0
  Download xclbin kaddr  : 0
  Download xclbin        : 0
  Reclock                : 0
  Peer data read         : 0
  User probe             : 0
  Mgmt state             : 0
  Change shell           : 0
  Reprogram shell        : 0
  P2P bar addr           : 0

Mechanical
  Fans
    FPGA Fan 1
      Critical Trigger Temp : 41 C
      Speed                 : 1208 RPM

Firewall
  Level 0: 0x0 (GOOD)

Thermals
  PCB Top Front          : 41 C
  PCB Top Rear           : 33 C
  PCB Bottom Front       : 39 C
  FPGA                   : 44 C

Qspi-status
  ERROR: xmc register base is not implemented on user windows

```

C:\Users\Administrator\Desktop\raj\XRT\build\WRelease\xilinx\xrt\bin>**xbmgmt examine -r all**
```
System Configuration
  OS Name              : Windows NT
  Release              : 20215.rs_prerelease.200911-1500
  Version              : 6.3
  Machine              : x86_64
  CPU Cores            : 6
  Memory               : 16382 MB
  Distribution         : Windows Server 2019 Datacenter
  Model                : Virtual Machine

XRT
  Version              : 2.12.0
  Branch               : 23jul_tmp
  Hash                 : 659a957a7fdeaf755120db6ffd8c2f948e64045b
  Hash Date            : 2021-07-23 20:56:39

Devices present
  [0000:00:00.0] : xilinx_u250_xdma_201830_2


-----------------------------------------------
1/1 [0000:00:00.0] : xilinx_u250_xdma_201830_2
-----------------------------------------------
Flash properties
  Type                 : spi
  Serial Number        : 21320493803J

Device properties
  Type                 : xilinx_u250_xdma_201830_2
  Name                 : AU250A64G
  Config Mode          : 7
  Max Power            : 75W

Flashable partitions running on FPGA
  Platform             : xilinx_u250_xdma_201830_2
  SC Version           : 4.2.0
  Platform ID          : 0x5d14fbe6

Flashable partitions installed in system
  Platform             : xilinx_u250_xdma_201830_2
  SC Version           : 4.2.0
  Platform ID          : 0x5d14fbe6


  Mac Address          : 00:0A:35:05:FE:28
                       : 00:0A:35:05:FE:29


Mechanical
  Fans
    FPGA Fan 1
      Critical Trigger Temp : 41 C
      Speed                 : 1208 RPM

Firewall
  Level --: -- --

Mailbox
  Total bytes received   : 3392 Bytes
  Unknown                : 0
  Test msg ready         : 0
  Test msg fetch         : 0
  Lock bitstream         : 0
  Unlock bitstream       : 0
  Hot reset              : 0
  Firewall trip          : 0
  Download xclbin kaddr  : 1
  Download xclbin        : 0
  Reclock                : 0
  Peer data read         : 26
  User probe             : 0
  Mgmt state             : 0
  Change shell           : 0
  Reprogram shell        : 0
  P2P bar addr           : 0
```

File: src/runtime_src/core/pcie/driver/windows/include/XoclMgmt_INTF.h [hunk 1]
```diff
@@ -109,12 +117,6 @@ typedef struct {
     CHAR   ShellFilename[64];
     CHAR   BMCVersion[16];
     UINT32 VlanTag;
-    CHAR   MacAddr0[32];
```
Comment (arpitxilinx) [inline src/runtime_src/core/pcie/driver/windows/include/XoclMgmt_INTF.h:112]: why was this removed? I am under the impression, that MS tools(azxbutil) was expecting this information. You might want to double check it.

Comment (rajkumar-xilinx) [reply src/runtime_src/core/pcie/driver/windows/include/XoclMgmt_INTF.h:112]: Yes, added MacAddress back to the struct. It was removed as part of https://github.com/Xilinx/XRT/pull/5476 actually. Now, fixed it here.

**Review by arpitxilinx — APPROVED** [review]

**Review by chvamshi-xilinx — APPROVED** [review]
